sensors/depth/depth.go

```diff
@@ -0,0 +1,53 @@
+// Package depth implements the Depth sensor
+package depth
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/registry"
+
+	"go.viam.com/slam/config"
+	"go.viam.com/slam/sensors"
+	"go.viam.com/slam/sensors/utils"
+)
+
+// Depth represents a depth sensor.
+type Depth struct {
```

Comment:
Do we need type validation that Depth inherits from Sensor?

Reply:
I think I don't understand the relationship between Sensor and Depth, since I would expect Depth to inherit from Sensor since Depth is a type of Sensor. I would expect us to have a slice of Sensors of different types, e.g. Depth and RGB, which allows us to call GetData on each different Sensor according to its own implementation. How are Sensor and Depth going to be used?

Reply:
Ah, I see.

Is anything else going to go into sensors.Sensor? If not, I think we don't need the type, since it is just a wrapper for an int Index. We could instead create a utility function to get the name from the index.

Reply:
I am not sure I understand. Maybe it's best to discuss this in person. The main point of `sensors.Sensor` is that we can define all sensors in a file such as this one: https://gist.github.com/kkufieta/4e225301ea97077517fb58450612fa1e#file-sensors-go

In the sensors.go file, we can clearly see the connection between the sensor (e.g. `lidar2D`, `imu`, etc) and the index. I am not sure how we could achieve the same type of human readability without using a `sensors.Sensor` type, but am very interested to hear about alternative approaches.

Originally, the plan was for `sensor.Sensor` to work with the idea of using a dict for sensors, and it contained an ID, a min & max data rate, and a default data rate (all of these have been removed due to the latest changes in scope).

Reply:
Recap from offline discussion: Will remove `sensors.Sensor` type and map variable names to indices directly, e.g. `lidar2D_index := 0`

Reply:
Done!